File: src/main/java/com/theonlytails/ruby/util/ClientEventBusSubscriber.java
package com.theonlytails.ruby.util;

import com.theonlytails.ruby.TheOnlyTails;
import com.theonlytails.ruby.clients.render.RubySheepRenderer;
import com.theonlytails.ruby.init.ItemsRegistry;
import com.theonlytails.ruby.init.RubyEntityTypes;
import com.theonlytails.ruby.items.RubySheepSpawnEgg;
import net.minecraft.block.DispenserBlock;
import net.minecraftforge.api.distmarker.Dist;
import net.minecraftforge.eventbus.api.SubscribeEvent;
import net.minecraftforge.fml.DeferredWorkQueue;
import net.minecraftforge.fml.client.registry.RenderingRegistry;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.event.lifecycle.FMLClientSetupEvent;
import net.minecraftforge.fml.event.lifecycle.FMLCommonSetupEvent;

@Mod.EventBusSubscriber(modid = TheOnlyTails.MOD_ID, bus = Mod.EventBusSubscriber.Bus.MOD, value = Dist.CLIENT)
public class ClientEventBusSubscriber {

    @SubscribeEvent
    public static void onClientSetup(FMLClientSetupEvent event) {
        RenderingRegistry.registerEntityRenderingHandler(RubyEntityTypes.RUBY_SHEEP.get(), RubySheepRenderer::new);
    }
}

File: src/main/kotlin/com/github/theonlytails/rubymod/util/ClientEventBusSubscriber.kt
package com.github.theonlytails.rubymod.util

import com.github.theonlytails.rubymod.RubyMod
import com.github.theonlytails.rubymod.items.CustomSpawnEggItem
import net.minecraft.entity.EntityType
import net.minecraftforge.api.distmarker.Dist
import net.minecraftforge.event.RegistryEvent
import net.minecraftforge.eventbus.api.SubscribeEvent
import net.minecraftforge.fml.common.Mod

@Mod.EventBusSubscriber(modid = RubyMod.MOD_ID, bus = Mod.EventBusSubscriber.Bus.FORGE, value = [Dist.CLIENT])
class ClientEventBusSubscriber {

	@SubscribeEvent
	fun onRegisterEntities(event: RegistryEvent.Register<EntityType<*>>) {
		CustomSpawnEggItem.initSpawnEggs()
	}
}